src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLoggerService.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Service responsible for logging versions
+    /// </summary>
+    public class FunctionsVersionLoggerService : IHostedService, IDisposable
+    {
+        private const double IntervalInSecondsDev = 1;    // interval for dev unit testing (1 s) in which the logs would be dumped
+        private readonly ILogger _logger;
+        private readonly IEnvironment _environment;
+        private readonly Timer _timer;
+        private readonly TimeSpan _interval;
+        private bool _disposed;
+        private double intervalInSecondsProd = 3600;      // default interval (1 hour) in which the logs would be dumped
+
+        public FunctionsVersionLoggerService(ILogger<FunctionsVersionLoggerService> logger, IEnvironment environment)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _interval = TimeSpan.FromSeconds(this.Interval);
+        }
+
+        public virtual double Interval
+        {
+            get { return (_environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName) == "development") ? IntervalInSecondsDev : intervalInSecondsProd; }
```

Comment:
I was wondering if we need to evaluate this environment variable every time we get the interval? And then I noticed that we only use this one in the constructor `TimeSpan.FromSeconds(this.Interval);`. From the point on we never use `Interval` as `_internal` is set.

Do we need to have a public property for interval? And since the `IntervalInSecondsDev` is for unit tests, I think it makes sense to refactor this so that we are evaluating the environment in the ctor and setting the `_interval` value there, and potentially eliminating the `Interval` prop